src/sentry/utils/kafka.py

```diff
@@ -65,7 +66,16 @@ def create_batching_kafka_consumer(topic_names, worker, **options):
             "Both 'force_topic' and 'force_cluster' have to be provided to override the configuration"
         )
     else:
-        cluster_names = {settings.KAFKA_TOPICS[topic_name]["cluster"] for topic_name in topic_names}
+        cluster_names: Set[str] = set()
+        for topic_name in topic_names:
+            cluster_name = settings.KAFKA_TOPICS[topic_name]
+            if region_name is not None:
```

Comment:
@RyanSkonnord @mikejihbe 

I'm assuming that in the end, the full list of region names and their mapping is a configuration that ops owns, and gets templated into `server.py` overrides.  To support minimal code divergence, I scoped atleast the kafka cluster config by an additional level (region name) which defaults to `"_local"` in the `sentry` repo, and would get templated in production.

I assume we can do a similar approach for other configurations, and that a master list of regions can be an additional `setting` option which also gets templated (and defaults to just `_local` otherwise).

Thoughts?

Reply:
That sounds reasonable enough for now. There's a good chance we'll have to rewrite a lot of this, however. One of my first steps would be to build some kind of object representation for the region -- the less we have to pass around `region_name` as a magic string, the better.

It seems like it's no longer convenient for the region-mapping config to be theoretical. I'll sketch out a skeleton some time this week, or we could pair on it if you want.

Reply:
I'm realizing now too that our use cases don't line up precisely. I do actually need to "leave" the magic string argument, because the region specific consumer configs are for the benefit of the control silo which consumes those regions  In other words, the region_name for consumer is not indicative of -this- region, but rather a -origin- region.

Reply:
This all simplifies away now.